GooglePlayInstant/Editor/PlayInstantLoadingScreenGenerator.cs

```diff
@@ -0,0 +1,93 @@
+// limitations under the License.
+
+using UnityEditor;
+using UnityEngine;
+using UnityEditor.SceneManagement;
+using UnityEngine.UI;
+using System.IO;
+
+namespace GooglePlayInstant.Editor
+{
+    public class PlayInstantLoadingScreenGenerator
+    {
+        public static void GenerateLoadingScreenScene(string pathToLoadingScreenImage, string assetBundleUrl)
+        {
+            if (!File.Exists(pathToLoadingScreenImage))
+            {
+                Debug.LogError("Loading screen image file cannot be found.");
+            }
+            else if (EditorSceneManager.GetSceneByName("instant-play-loading-screen-scene").IsValid())
+            {
+                Debug.LogError("A generated loading scene already exists.");
+            }
+            else
+            {
+                var loadingScreenScene = EditorSceneManager.NewScene(NewSceneSetup.EmptyScene, NewSceneMode.Additive);
+                var loadingScreenGO = new GameObject("Canvas");
+            
+                GenerateLoadingScreenScript(assetBundleUrl);
+                AddBackgroundImageToScene(loadingScreenGO, pathToLoadingScreenImage);
+                AddLoadingScreenScript(loadingScreenGO);
+
+                EditorSceneManager.SaveScene(loadingScreenScene, "instant-play-loading-screen-scene.unity");
+            }
+        }
+        
+        //TODO: get rid of error message associated with using the LoadingScreenScript reference
+        private static void AddLoadingScreenScript(GameObject loadingScreenGO)
+        {
+            loadingScreenGO.AddComponent<LoadingScreenScript>();
+        }
+        
+        private static void AddBackgroundImageToScene(GameObject loadingScreenGO, string pathToLoadingScreenImage)
+        {
+            loadingScreenGO.AddComponent<Canvas>();
+            var loadingScreenCanvas = loadingScreenGO.GetComponent<Canvas>();
+            loadingScreenCanvas.renderMode = RenderMode.ScreenSpaceOverlay;
+            
+            byte[] loadingScreenImageData = File.ReadAllBytes(pathToLoadingScreenImage);
+            var tex = new Texture2D(1, 1);
+            tex.LoadImage(loadingScreenImageData);
+
+            var loadingImageSprite = Sprite.Create(tex, new Rect(0, 0, tex.width, tex.height), new Vector2(0.5f, 0.5f));
+
+            loadingScreenGO.AddComponent<Image>();
+            var loadingScreenImage = loadingScreenGO.GetComponent<Image>();
+            loadingScreenImage.sprite = loadingImageSprite;
+        }
+
+        //TODO: add better handling of finding assets folder and figure out possible alternative AssetDatabase synchronous importing
+        private static void GenerateLoadingScreenScript(string assetBundleUrl)
+        {
+            var newloadingScreenScriptDir = Directory.GetCurrentDirectory() + "/Assets/GooglePlayInstantScript/LoadingScreenScript.cs";
+            if (Directory.GetFiles(Directory.GetCurrentDirectory(), "GenericLoadingScreenScript.cs",
```

Comment:
Extract a local variable for
Directory.GetFiles(Directory.GetCurrentDirectory(), "GenericLoadingScreenScript.cs", SearchOption.AllDirectories)

use this in place of the 2 calls below.

Reply:
Resolved.